packages/react-components/src/components/table-output-component.tsx

```diff
@@ -83,13 +84,13 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
                     this.fetchColumns = false;
                     await this.init();
                 }
-                const rowsThisPage = await this.fetchSearchTableLines(params.startRow, params.endRow - params.startRow);
+                const rowsThisPage = await this.fetchTableLines(params.startRow, params.endRow - params.startRow);
                 for (let i = 0; i < rowsThisPage.length; i++) {
                     const item = rowsThisPage[i];
                     const itemCopy = cloneDeep(item);
                     rowsThisPage[i] = itemCopy;
                 }
-                params.successCallback(rowsThisPage, this.props.nbEvents);
+                params.successCallback(rowsThisPage, this.tableSize);
```

Comment:
It took me a while to understand why it was working on with events table and not with the segment store table. For the events table there is an indexing loop that checks for number of events and updates the nbEvents which then triggers the function call componentDidUpdate() on line 158 below. The componentDidUpdate() sets the infiniteRowCount which then updates the vertical scrollbar. When opening the table with a segment store there is no corresponding indexing loop that updates the nbEvents and the scrollbar.

The fix that you suggest, will work, only when the fetchTableLines call to fill the table initially returns the total number of segments i.e. that the back-end has the value available at that time. 

I think there should be a generic way to determine the table size for any virtual table data providers. I'll open a separate ticket for that. I'll merge this PR because it works-around the bug for segment store tables and it still works with the events table.